src/radical/entk/appman/appmanager.py

```diff
@@ -203,7 +252,11 @@ def run(self):
 
             else:
 
+                with open('entk_{}.json'.format(self._uid), 'w') as outfile:  
+                    json.dump(self._data['resource dictionary'].append(self._resource_manager), outfile)
+
                 self._prof.prof('amgr run started', uid=self._uid)
+                self._resource_manager
```

Comment:
The fact that you use an empty attribute statement here suggests that the json dump should be moved to a separate, private (or public?) method.  Attribute statements are for inspection, and should have minimal side effects.

Reply:
i still don't understand why this is implemented as a side effect to an attribute.

Consider the following (made-up) example:

```
while  self._resource_manager.check_state():
  print 'ok'
```
This would, at every iteration of that loop, save the json data set, fo no reason whatsoever.  The implementation basically hijacks the attribute.  Why not move the json dump to, well, something like `self._dump_json()` or whatever, which is called when the intent is to dump the json files?